### PR TITLE
PTV-1836: Update to python3 and fix file load error

### LINKED ISSUE #8: Update to python 3
Still on python 2.7.6

### PR DESCRIPTION
**Pull request by MrCreosote**
Python2.7 is so old now everything breaks, needed to update just to build

The filtered file produced by AssemblyUtil moved position, update to use the location returned by the method

Closes #8

### CODE REVIEW

**Comment by MrCreosote** [comment]
Almost all of the 6k lines is new generated clients clients

File: lib/kb_IDBA/kb_IDBAImpl.py [hunk 1]
```diff
@@ -459,13 +459,15 @@ def run_idba_ud(self, ctx, params):
...
+            filtered_file = assemblyUtil.save_assembly_from_fasta2(
+                {
+                    'file': {'path': output_contigs},
+                    'workspace_name': wsname,
+                    'assembly_name': params[self.PARAM_IN_CS_NAME],
+                    'min_contig_length': params['min_contig_length']
+                })['filtered_input']
             # load report from scaffolds.fasta
-            report_name, report_ref = self.load_report(output_contigs+'.filtered.fa', params, wsname)
+            report_name, report_ref = self.load_report(filtered_file, params, wsname)
```
Comment (MrCreosote) [inline lib/kb_IDBA/kb_IDBAImpl.py:470]: This is the actual bugfix

File: test/kb_IDBA_server_test.py [hunk 2]
```diff
@@ -91,59 +96,19 @@ def delete_shock_node(cls, node_id):
                         allow_redirects=True)
         print('Deleted shock node ' + node_id)
 
-
-    # Helper script borrowed from the transform service, logger removed
```
Comment (MrCreosote) [inline test/kb_IDBA_server_test.py:95]: had to dig out this archaeological specimen

**Comment by dcchivian** [comment]
@MrCreosote are unit tests too big for github actions to run?  we should have github actions unit tests run.

**Comment by MrCreosote** [comment]
I don't know, but I'm not going to add GHA tests as part of the PR; it's easy enough for anyone that wants to verify to clone and run tests. I did add an issue to add GHA tests at some point in the future: #9

**Comment by dcchivian** [comment]
I was under the impression that GHA was part of our code merge protocol now.  It's trivial to add.

**Comment by dcchivian** [comment]
how about this?  if you tell me that unit tests all passed, then we'll add the GHA as a separate PR.

**Comment by MrCreosote** [comment]
I'm unaware of any official merge protocol, but if there is one I'm obeying the spirit if not the letter:
```
~/github/kbaseapps/kb_IDBA$ kb-sdk test

Validating module in (/home/crushingismybusiness/github/kbaseapps/kb_IDBA)
*snip*
=============== Cleaning up ==================
Test workspace was deleted: test_kb_IDBA1665686557726
Deleted shock node c1de1d8b-e8c2-41ad-a57d-394fce044cfd
Deleted shock node cee6e062-a284-4802-bc8d-d506e8a47781
Deleted shock node 61544c1a-a6c2-458c-94df-932808ba6387
Deleted shock node b239190e-6f32-4067-b6b9-1f12e171fc2a
...............
----------------------------------------------------------------------
Ran 15 tests in 218.926s

OK
Deleted handles ['KBH_700830', 'KBH_700831', 'KBH_700833']
Shutting down callback server...
1665686777.07 - CallbackServer: Shutting down executor service
```

**Review by dcchivian — APPROVED** [review]
changes look good.  we will add GHA tests in next PR

**Comment by MrCreosote** [comment]
```
~/github/kbaseapps/kb_IDBA$ git log
commit 6a297ea5afa4b50cfdbbbe81c2b700f05f290290 (HEAD -> dev-update_python, origin/dev-update_python)
Author: Gavin <gaprice@lbl.gov>
Date:   Wed Oct 12 12:07:19 2022 -0700

    Reompile for py3 compatibiilty
    
    The tests don't actually test the server
```

**Pull request merged**